apps/btc_family/btc_txn.c

```diff
@@ -247,31 +286,175 @@ static bool fetch_transaction_meta(btc_query_t *query) {
       !check_which_request(query, BTC_SIGN_TXN_REQUEST_META_TAG)) {
     return false;
   }
-  // TODO: add checks for validating the metadata
+  // check important information for supported/compatibility
+  if (0x00000001 != query->sign_txn.meta.sighash ||
+      0 == query->sign_txn.meta.input_count ||
+      0 == query->sign_txn.meta.output_count) {
```

Comment:
We were also thinking of having some upper bound checks on input and output count.. right?

Can we add a TODO here?

Reply:
added